src/firmware/RC_firmware/RC_firmware.ino

```diff
@@ -0,0 +1,233 @@
+/* Firmware for Elsa
+   Test version
+   Author: Vincent Yuan, Modified: Nick Wu
+     Modified: James Asefa
+   Date: Oct 30, 2016
+*/
+
+#include <SoftwareSerial.h>
+#include <stdlib.h>
+#include <Servo.h>
+
+
+#define TRIM 8 // error margin for joysticks
+
+#define BAUD_RATE 115200
+#define BUFFER_SIZE 6
+// assign 128 to be not moving
+// will be used to hold robot in place while
+// reading from the buffer. Robot will be stopped initially.
+#define UNMAPPED_STOP_SPEED 128
+
+// DRIVING MODIFICATIONS
+// modify FR, FL, BR, BL to change speeds: (+) correpsonds to faster speed, (-) for going reverse
+// F:FRONT, B:BACK, R:RIGHT, L:LEFT e.g. FR = FRONT RIGHT WHEEL.
+const int OFFSET = 1;
+const int joystick_margin = 150;
+
+// buffer inputs
+int linear_x = 0;
+int linear_y = 0;
+int linear_z = 0;
+int angular_x = 0;
+int angular_y = 0;
+int angular_z = 0;
+
+// motor pins
+const int left_motor_pin = 9;
+const int right_motor_pin = 10;
+// defines start of buffer
+const char buffer_head = 'B';
+
+
+// max and min linear speeds and stopping condition
+const int linear_max = 100;
+const int linear_min = 80;
+const int linear_stop = 90;
+
+// max and min angular speeds and stopping condition
+const int angular_max = 100;
+const int angular_min = 80;
+const int angular_stop = 90;
+
+Servo LeftM;
+Servo RightM;
+
+
+// these variables will store the joystick ranges - used to figure out direction, turn etc
+int range1 = 0, range2 = 0, range3 = 0, range4 = 0, B2 = 0, B4 = 0, Mode = 0;
+
+int linearXHigh = 0, linearXLow = 0, angularZHigh = 0, angularZLow = 0;
+
+void setup() {
+  Serial.begin(BAUD_RATE);
+  
+  pinMode(2, INPUT);
+  pinMode(3, INPUT);
+  pinMode(4, INPUT);
+  pinMode(5, INPUT);
+  
+  LeftM.attach(left_motor_pin);
+  RightM.attach(right_motor_pin);
+  
+  // calculate offset
+  set_off();
+}
+
+void loop() {
+
+  //if (Serial.read() == 'I') {Serial.print("DRIVE");}
+  //else Serial.flushRX();
+  //responding to queries from usb
```

Comment:
Commented out code should be removed, part of the reason for using a version control system is that you have access to all your old code, but it keeps the present iteration of your code clean.